documentation/customisation.md

```diff
@@ -0,0 +1,139 @@
+# Dockerfile Customisation
+
+A custom Docker file might be required in certain circumstances. These include (but aren't limited to)
+
+- Connecting to a properietary database (DB2, Oracle, etc.)
+- Adding a hotfix as instructed by our support team
+- Adding a custom plugin
+- etc.
+
+The process to customise a docker image for your needs is described below
+
+## XL Deploy
+
+When creating a Dockerfile with custom resources added, always remember that the owning user:group combination _must_ be `10001:0`, or else XL Deploy will not be able to read the files. 
+
+To start off with, create a `Dockerfile` that resembles the following. 
+
+Please note that certain JARS will go into certain paths. You shouldn't add a Oracle JAR to the `ext/` folder, for example. If you are unsure where a JAR should be added, please get in touch with our support team. 
+
+NOTE: `${APP_HOME}` points to the path `/opt/xebialabs/xl-deploy-server` by default
+
+```docker
+FROM xebialabs/xl-deploy:9.5.0
+
+###################################################################################
+# PLUGINS                                                                         #
+# Plugins should be placed under ${APP_HOME}/default-plugins/ #
+###################################################################################
+
+COPY --chown=10001:0 files/xld-liquibase-plugin-5.0.1.xldp /opt/xebialabs/xl-deploy-server/default-plugins/
+
+# Add plugin from url
+ADD --chown=10001:0 https://dist.xebialabs.com/public/community/xl-deploy/command2-plugin/3.9.1-1/command2-plugin-3.9.1-1.jar /opt/xebialabs/xl-deploy-server/default-plugins/
```

Comment:
- This is really old version of command2 plugin since 2015 .. better to use latest one below,
https://github.com/xebialabs-community/xld-advanced-command-plugin/releases/download/v3.2.4/xld-advanced-command-plugin-3.2.4.xldp

Reply:
this is an example, user can use whatever plugin user wants